model/apmevent.go

```diff
@@ -124,7 +128,9 @@ func (e *APMEvent) BeatEvent(ctx context.Context) beat.Event {
 	fields.maybeSetMapStr("service", e.Service.Fields())
 	fields.maybeSetMapStr("agent", e.Agent.fields())
 	fields.maybeSetMapStr("observer", e.Observer.Fields())
-	fields.maybeSetMapStr("host", e.Host.fields())
+	if e.Firehose != nil {
+		fields.maybeSetMapStr("host", e.Host.fields())
+	}
```

Comment:
Please remove this conditional logic. We should not be setting e.Host. If we don't set it, then it won't be emitted in the doc. I think if you remove the request metadata function from your handler, this won't be necessary.